modules/uploads/utils/types.ts

```diff
@@ -0,0 +1,87 @@
+import {
+	Files as PrismaFiles,
+	Upload as _PrismaUpload,
+} from "../_gen/prisma/default.d.ts";
+
+interface PrismaUpload extends Omit<_PrismaUpload, "deletedAt"> {
+	files: PrismaFiles[];
+}
+
+export interface Upload {
+	id: string;
+	userId: string | null;
+
+	bucket: string;
+	contentLength: string;
+
+	files: UploadFile[];
+
+	createdAt: string;
+	updatedAt: string;
+	completedAt: string | null;
+}
+
+export interface UploadFile {
+	path: string;
+	mime: string | null;
+	contentLength: string;
+}
+
+export interface PresignedUpload extends Omit<Upload, "files"> {
+	files: PresignedUploadFile[];
+}
+
+export interface PresignedUploadFile extends UploadFile {
+	presignedChunks: PresignedChunk[];
+}
+
+export interface PresignedChunk {
+	url: string;
+
+	partNumber: number;
+	contentLength: string;
+	offset: string;
+}
+
+export type UploadWithOptionalFiles =
+	& Omit<Upload, "files">
+	& Partial<Pick<Upload, "files">>;
+export type PrismaUploadWithOptionalFiles =
+	& Omit<PrismaUpload, "files">
+	& Partial<Pick<PrismaUpload, "files">>;
```

Comment:
simplify these types, less typescript-ing. focus on legibility for people who aren't typescript phds.